src/drivers/frsky_telemetry/frsky_telemetry.c

```diff
@@ -195,26 +194,33 @@ static int frsky_telemetry_thread_main(int argc, char *argv[])
 	char sbuf[20];
 
 	/* 2 byte polling frames indicate SmartPort telemetry
-	 * 11 byte packets, indicate D type telemetry
+	 * 11 byte packets indicate D type telemetry
 	 */
 	int status = poll(fds, sizeof(fds) / sizeof(fds[0]), 3000);
 
-//	warnx("poll status: %u", status);
 	if (status > 0) {
 		/* received some data; check size of packet */
 		usleep(5000);
 		status = read(uart, &sbuf[0], sizeof(sbuf));
-//		warnx("received %u bytes", status);
 	}
 
 	if (status > 0 && status < 3) {
 		/* traffic on the port, D type is 11 bytes per frame, SmartPort is only 2 */
 		/* Subscribe to topics */
-		sPort_init();
+		if (!sPort_init()) {
+			err(1, "could not allocate memory");
+		}
 
 		warnx("sending FrSky SmartPort telemetry");
 
-		int sensor_sub = orb_subscribe(ORB_ID(sensor_combined));
+		struct sensor_baro_s *sensor_baro = malloc(sizeof(struct sensor_baro_s));
+
+		if (sensor_baro == NULL) {
+			err(1, "could not allocate memory");
+		}
+
+		static float filtered_alt = NAN;
```

Comment:
Not necessarily required to change, but just FYI: This float will also be statically allocated.